testsuite/MDAnalysisTests/formats/test_libdcd.py

```diff
@@ -0,0 +1,551 @@
+                box=frame.unitcell.astype(np.float64)
+                f_out.write(xyz=frame.x,
+                            box=box,
+                            step=f_in.istart,
+                            natoms=frame.x.shape[0],
+                            charmm=1, # DCD should be CHARMM
+                            time_step=f_in.delta,
+                            ts_between_saves=f_in.nsavc,
+                            remarks=remarks)
+
+    def tearDown(self):
+        try:
+            os.unlink(self.testfile)
+        except OSError:
+            pass
+        del self.tmpdir
+
+    @raises(IOError)
+    def test_write_mode(self):
+        # ensure that writing of DCD files only occurs with properly
+        # opened files
+        with DCDFile(self.readfile) as dcd:
+            dcd.write(xyz=np.zeros((3,3)),
+                      box=np.zeros(6, dtype=np.float64),
+                      step=0,
+                      natoms=330,
+                      charmm=0,
+                      time_step=22.2,
+                      ts_between_saves=3,
+                      remarks='')
+
+    def test_written_dcd_coordinate_data_shape(self):
+        # written coord shape should match for all frames
+        expected = (self.natoms, 3)
+        with DCDFile(self.testfile) as f:
+            if f.n_frames > 1:
+                for frame in f:
+                    xyz = f.read()[0]
+                    assert_equal(xyz.shape, expected)
+            else:
+                xyz = f.read()[0]
+                assert_equal(xyz.shape, expected)
+
+    def test_written_unit_cell(self):
+        # written unit cell dimensions should match for all frames
+        with DCDFile(self.testfile) as test, DCDFile(self.readfile) as ref:
+            curr_frame = 0
+            while curr_frame < test.n_frames:
+                written_unitcell = test.read()[1]
+                ref_unitcell = ref.read()[1]
+                curr_frame += 1
+                assert_equal(written_unitcell, ref_unitcell)
+
+    def test_written_num_frames(self):
+        with DCDFile(self.testfile) as f:
+            assert_equal(len(f), self.expected_frames)
+
+    def test_written_seek(self):
+        # ensure that we can seek properly on written DCD file
+        with DCDFile(self.testfile) as f:
+            f.seek(self.seek_frame)
+            assert_equal(f.tell(), self.seek_frame)
+
+    def test_written_zero_based_frames(self):
+        # ensure that the first written DCD frame is 0
+        expected_frame = 0
+        with DCDFile(self.testfile) as f:
+            assert_equal(f.tell(), expected_frame)
+
+    def test_written_remarks(self):
+        # ensure that the REMARKS field *can be* preserved exactly
+        # in the written DCD file
+        with DCDFile(self.testfile) as f:
+            assert_equal(f.remarks, self.expected_remarks)
+
+    @given(st.text()) # handle the full unicode range of strings
+    def test_written_remarks_property(self, remarks_str):
```

Comment:
@tylerjereddy I will deactivate these tests for now. The reason is that hypothesis also returns unprintable unicode characters that are striped in the libdcd wrapper when we read remarks from a file. This seems to have fixed some problems for you with string parsing. You can activate them yourself again later.

I also didn't find how I could quickly tell hypothesis to restrict the text to printable unicode symbols.

btw all strings you look at with this function should use `print(repr(val))` during debug because val itself might not be printable.